source/s3_client.c

```diff
@@ -365,6 +359,14 @@ struct aws_s3_client *aws_s3_client_new(
         }
     }
 
+    if (client_config->proxy_options) {
+        client->proxy_config = aws_http_proxy_config_new_from_proxy_options_with_tls_info(
```

Comment:
Thanks, we can't handle that in binding because we will have to handle it in every binding. This code is already being used by sdk-cpp. I will document this.